src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -54,8 +54,15 @@ type mgmtModule struct {
 	monitor    *procMon
 }
 
+func isManagementMethod(method drpc.Method) bool {
+	return method == drpc.MethodGetAttachInfo ||
+		method == drpc.MethodDisconnect ||
+		method == drpc.MethodPoolConnect ||
+		method == drpc.MethodPoolDisconnect
+}
+
 func (mod *mgmtModule) HandleCall(session *drpc.Session, method drpc.Method, req []byte) ([]byte, error) {
-	if method != drpc.MethodGetAttachInfo && method != drpc.MethodDisconnect {
+	if !isManagementMethod(method) {
 		return nil, drpc.UnknownMethodFailure()
 	}
```

Comment:
I'm still in favor of getting rid of this check altogether. We should get there anyway after going through the switch, and that way we wouldn't need to duplicate the list of methods handled by the module.

Reply:
@jbrosenz Since you originally put this check in what is your opinion on the premature exit if its not the right method?

Reply:
I didn't author the original function, but I did make edits.  That said, I have no particular problem with the way you have it structured currently.  If you do get rid of the check, you could add a default case to the switch, that returns nil, drpc.UnknownMethodFailure().  I don't think it's a particularly likely problem that an unknown method will be used, so the penalty of executing the code before it, prior to exiting, is not really a big deal.  If there is an unhandled method, it means there's a version mismatch.  Your code won't break, and the appropriate error will be returned that shows there was a mismatch.

Reply:
Removed the check since everyone is in agreement.